pkg/probe/exec/exec_test.go

```diff
@@ -110,25 +110,26 @@ func TestExec(t *testing.T) {
 		expectedStatus probe.Result
 		expectError    bool
 		input          string
+		stdout         string
 		output         string
 		err            error
 	}{
 		// Ok
-		{probe.Success, false, "OK", "OK", nil},
+		{probe.Success, false, "OK", "OK", "OK", nil},
 		// Ok
-		{probe.Success, false, "OK", "OK", &fakeExitError{true, 0}},
+		{probe.Success, false, "OK", "OK", "OK", &fakeExitError{true, 0}},
 		// Ok - truncated output
-		{probe.Success, false, elevenKilobyte, tenKilobyte, nil},
+		{probe.Success, false, elevenKilobyte, tenKilobyte, tenKilobyte, nil},
 		// Run returns error
-		{probe.Unknown, true, "", "", fmt.Errorf("test error")},
+		{probe.Unknown, true, "", "", "", fmt.Errorf("test error")},
 		// Unhealthy
-		{probe.Failure, false, "Fail", "", &fakeExitError{true, 1}},
+		{probe.Failure, false, "Fail", "", "exitCode: 1, stderr: fake exit, stdout: ", &fakeExitError{true, 1}},
 		// Timeout
-		{probe.Failure, false, "", "command testcmd timed out", NewTimeoutError(fmt.Errorf("command testcmd timed out"), time.Second)},
+		{probe.Failure, false, "", "", "command testcmd timed out", NewTimeoutError(fmt.Errorf("command testcmd timed out"), time.Second)},
 	}
 	for i, test := range tests {
 		fake := FakeCmd{
-			out: []byte(test.output),
+			out: []byte(test.stdout),
 			err: test.err,
 		}
```

Comment:
It would be nice to write these tests with real commands, to see what the actual output would look like. That probably requires isolating the tests per OS with build tags though.

Reply:
yeah, I'm willing to do this, but I found the process exist error msg is  closely related to the error output of the user script, thus I pay more attention to the correctness of the program results in the test and updated the Exec test method to make it clearer and more comprehensive.